examples/cache/src/main/java/momento/client/example/BatchUtilsExample.java

```diff
@@ -0,0 +1,106 @@
+package momento.client.example;
+
+import java.time.Duration;
+import java.util.Arrays;
+import momento.sdk.CacheClient;
+import momento.sdk.auth.CredentialProvider;
+import momento.sdk.auth.EnvVarCredentialProvider;
+import momento.sdk.batchutils.MomentoBatchUtils;
+import momento.sdk.batchutils.request.BatchGetRequest;
+import momento.sdk.batchutils.response.BatchGetResponse;
+import momento.sdk.config.Configurations;
+import momento.sdk.exceptions.AlreadyExistsException;
+import momento.sdk.responses.cache.GetResponse;
+import momento.sdk.responses.cache.control.CacheCreateResponse;
+
+/**
+ * BatchUtilsExample demonstrates the use of MomentoBatchUtils for performing batch operations with
+ * the Momento SDK. This class includes examples of setting up a cache, adding test data, and
+ * performing batch get operations.
+ *
+ * <p>The example covers: - Creating a cache if it does not exist. - Adding a set of key-value pairs
+ * to the cache. - Performing a batch get operation to retrieve multiple keys in a single call. -
+ * Handling responses and displaying the results.
+ *
+ * <p>It utilizes the Momento SDK's CacheClient for cache operations and the MomentoBatchUtils for
+ * handling batch operations efficiently.
+ */
+public class BatchUtilsExample {
+
+  private static final String API_KEY_ENV_VAR = "MOMENTO_API_KEY";
+  private static final Duration DEFAULT_ITEM_TTL = Duration.ofSeconds(60);
+
+  private static final String CACHE_NAME = "cache";
+
+  public static void main(String[] args) {
+
+    final CredentialProvider credentialProvider = new EnvVarCredentialProvider(API_KEY_ENV_VAR);
+
+    try (final CacheClient client =
+        CacheClient.create(credentialProvider, Configurations.Laptop.latest(), DEFAULT_ITEM_TTL)) {
+
+      createCache(client, CACHE_NAME);
+      printStartBanner();
+
+      try (final MomentoBatchUtils momentoBatchUtils = MomentoBatchUtils.builder(client).build()) {
+        setupTestData(client, CACHE_NAME);
+        performBatchGet(momentoBatchUtils, CACHE_NAME);
+      }
```

Comment:
At first glance I had a hard time following where the batch builder was instantiated; I was looking for it in the perform batch get method. That said I understand that you've organized like this because it's part of the "setup" code.

Suggestions:
1. Could you add comment in that method with a reference to this block for instantiating it, and also mention the use of the `try` block because the builder is `Closeable`?
2. Make explicit the max number of concurrent requests as a member field and the number of elements to add and fetch. Eg so I can try adding 10 elements and fetching with a concurrency of 5 vs 2.

Reply:
yeah usually i like the style with `builder` intended on the new line, but our spotless configs change that. Will address these